app/code/Magento/Quote/Model/QuoteManagement.php

```diff
@@ -298,22 +298,30 @@ public function assignCustomer($cartId, $customerId, $storeId)
             );
         }
         try {
-            $this->quoteRepository->getForCustomer($customerId);
-            throw new StateException(
-                __("The customer can't be assigned to the cart because the customer already has an active cart.")
-            );
-        // phpcs:ignore Magento2.CodeAnalysis.EmptyBlock
+            $customerActiveQuote = $this->quoteRepository->getForCustomer($customerId);
         } catch (\Magento\Framework\Exception\NoSuchEntityException $e) {
+            /** This exception appear when customer have no active cart*/
+            $customerActiveQuote = false;
         }
 
+        if ($customerActiveQuote) {
+            /** Merge carts */
+            $quote->merge($customerActiveQuote);
+            $this->quoteRepository->delete($customerActiveQuote);
+        }
         $quote->setCustomer($customer);
         $quote->setCustomerIsGuest(0);
+        $quote->setStoreId($storeId);
```

Comment:
What the reason to set storeid?

Reply:
@VladimirZaets , I have tried to use data from method parameters to be sure that right store_id for quote was used. I see that we have checking if store_id is valid in this method and we have this value defined in quote record. store_id value assignment removed